engine/time/src/main/java/io/deephaven/time/DateTimeUtils.java

```diff
@@ -3886,6 +3886,11 @@ public static Instant parseInstant(@NotNull final String s) {
                 return epochNanosToInstant(nanos);
             }
 
+            if (s.startsWith("Z") || s.startsWith("z")) {
+                throw new RuntimeException(
+                        "A 'Z' or 'z' prefix is not allowed for Instant parsing.  Use parseZonedDateTime instead.");
+            }
```

Comment:
I'm not a fan of the way this is structured implementation-wise; also, `parseEpochNanos` delegates to parseZonedDateTime - I'm not sure if we _want_ parseEpochNanos to accept `Z` prefix, but if we do, it should have javadoc updated too.

I think it would be better to have a `parseZonedDateTimeImpl` that is private and _doesn't_ accept `Z`, and existing callers (here and maybe parseEpochNanos) can instead call the Impl version.